tests/Opus/Doi/DataCiteXmlGeneratorTest.php

```diff
@@ -139,4 +139,171 @@ public function testGetStylesheetPathConfigured()
 
         $this->assertEquals($path, $stylesheetPath);
     }
+
+    /**
+     * The DataCite-XML should not contain files, which are invisible in oai
+     * Test if both invisible files are hided
+     */
+    public function testFileInformationInvisible()
+    {
+        $doc = new Opus_Document();
+        $this->addRequiredPropsToDoc($doc);
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(0);
+        $file->setFileSize('0');
+        $file->setMimeType('pdf');
+        $doc->addFile($file);
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(0);
+        $file->setFileSize('0');
+        $file->setMimeType('pdf');
+        $doc->addFile($file);
+
+        $generator = new Opus_Doi_DataCiteXmlGenerator();
+        $result = $generator->getXml($doc);
+
+        $this->assertNotContains('<sizes><size>0 KB</size></sizes>', $result);
+        $this->assertNotContains('<formats><format>pdf</format></formats>', $result);
+    }
+
+    /**
+     * Creates a txt-file with random size.
+     *
+     * @return string path of file
+     * @throws Zend_Exception
+     */
+    private function createTestFile() {
+        $config = Zend_Registry::get('Zend_Config');
+        $path = $config->workspacePath . DIRECTORY_SEPARATOR . uniqid();
+
+        $src_path = $path . DIRECTORY_SEPARATOR . 'src';
+        mkdir($src_path, 0777, true);
+
+        $filename_nonzero = $src_path . DIRECTORY_SEPARATOR . 'foobar-nonzero.txt';
+        $fh = fopen($filename_nonzero, 'w');
+
+        if ($fh == false) {
+            $this->fail("Unable to write file $filename_nonzero.");
+        }
+
+        $rand = rand(1000, 100000);
+        for ($i = 0; $i < $rand; $i++) {
+            fwrite($fh, ".");
+        }
+
+        fclose($fh);
+
+        return $filename_nonzero;
+    }
+
+    /**
+     * The DataCite-XML should not contain files, which are invisible in oai
+     * Test if both visible files are shown
+     */
+    public function testFileInformationVisible()
+    {
+        $doc = new Opus_Document();
+        $this->addRequiredPropsToDoc($doc);
+
+        $filename = $this->createTestFile();
+
+        $file = New Opus_File();
+        $file->setVisibleInOai(1);
+        $file->setTempFile($filename);
+        $file->setPathName('copied-foobar-nonzero.txt');
+        $doc->addFile($file);
+        $doc->store();
+
+        $filename2 = $this->createTestFile();
+
+        $file2 = New Opus_File();
+        $file2->setVisibleInOai(1);
+        $file2->setTempFile($filename2);
+        $file2->setPathName('copied-foobar-nonzero_2.txt');
+        $doc->addFile($file2);
+        $doc->store();
+
+        $generator = new Opus_Doi_DataCiteXmlGenerator();
+        $result = $generator->getXml($doc);
+
+        $size = intval(round($file->getFileSize()/1024));
+        $size2 = intval(round($file2->getFileSize()/1024));
+
+        $this->assertContains("<sizes><size>$size KB</size><size>$size2 KB</size></sizes>", $result);
```

Comment:
Sind die Testdateien nicht leer, so dass hier weiterhin "0 KB" steht und unklar ist, ob eine korrekte Ausgabe erfolgt und die korrekte Datei getestet wird?

Reply:
Nein. Es werden Testfiles erstellt, die eine zufällige Größe haben. Siehe Funktion _createTestFile_. Eine ähnliche findet sich in _FileTest_.